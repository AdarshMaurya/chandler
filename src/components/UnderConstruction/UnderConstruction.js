import React, {PureComponent} from 'react';
import {View, Text, Image} from 'react-native';

import {commonStyles} from 'theme';
import {styles} from './UnderConstruction.styles';

export class UnderConstruction extends PureComponent {
  render() {
    return (
      <View style={styles.flexCentered}>
        <View style={commonStyles.flex} />
        <Image source={require('./images/under-construction.png')} />
        <Text style={styles.text}>This page is under construction</Text>
        <View style={commonStyles.flex} />
      </View>
    );
  }
}
