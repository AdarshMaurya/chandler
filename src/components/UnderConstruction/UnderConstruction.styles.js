import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  text: {
    textAlign: 'center',
    fontSize: 18,
    marginTop: 30,
    paddingHorizontal: '20%',
  },
  flexCentered: {
    flex: 1,
    alignItems: 'center',
  },
});
