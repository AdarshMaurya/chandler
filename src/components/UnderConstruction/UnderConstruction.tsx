import React, {PureComponent} from 'react';
import {Image, Text, View} from 'react-native';

import {commonStyles} from '@theme';
import {styles} from './UnderConstruction.styles';

export class UnderConstruction extends PureComponent<{}, {}> {
  public render() {
    return (
      <View style={styles.flexCentered}>
        <View style={commonStyles.flex} />
        <Image source={require('./images/under-construction.png')} />
        <Text style={styles.text}>This page is under construction</Text>
        <View style={commonStyles.flex} />
      </View>
    );
  }
}
