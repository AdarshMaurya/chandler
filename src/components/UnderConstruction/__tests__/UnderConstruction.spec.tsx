import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {UnderConstruction} from '../UnderConstruction';

describe('Components', () => {
  describe('UnderConstruction', () => {
    it('should renders correctly', () => {
      expect(toJson(shallow(<UnderConstruction />))).toMatchSnapshot();
    });
  });
});
