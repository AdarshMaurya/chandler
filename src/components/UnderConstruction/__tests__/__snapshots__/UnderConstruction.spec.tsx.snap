// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components UnderConstruction should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Image
    source={1}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "marginTop": 30,
        "paddingHorizontal": "20%",
        "textAlign": "center",
      }
    }
  >
    This page is under construction.
  </Text>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
